base/steer/FairWriteoutBuffer.cxx

```diff
@@ -14,256 +14,269 @@
-    delete dataClone;
-  }
-
 }
 
 void FairWriteoutBuffer::FillDataToDeadTimeMap(FairTimeStamp* data, double activeTime, double startTime)
 {
-  if (fActivateBuffering) {
-    typedef  std::multimap<double, FairTimeStamp*>::iterator DTMapIter;
-//    typedef  std::map<FairTimeStamp, double>::iterator DataMapIter;
-    if (activeTime < 0) activeTime = 0; // to avoid errors due to wrong (negative) returnvalues of overwritten modify function by subdetector groups. A negative (-1) aktiveTime already indicate empty data maps!
-    double timeOfOldData = FindTimeForData(data);
-    // PrintDeadTimeMap();
-    if(timeOfOldData > -1) {        //if an older active data object is already present
-      if (fVerbose > 1) {
-        std::cout << " OldData found! " << timeOfOldData << std::endl;
-      }
-      if (fVerbose > 1) {
-        std::cout << "New Data: " << activeTime << " : " << data << std::endl;
-      }
-      double currentdeadtime = timeOfOldData;
-      FairTimeStamp* oldData;
-      bool dataFound = false;
+    if (fActivateBuffering) {
+        typedef std::multimap<double, FairTimeStamp*>::iterator DTMapIter;
+        //    typedef  std::map<FairTimeStamp, double>::iterator DataMapIter;
+        if (activeTime < 0)
+            activeTime = 0;   // to avoid errors due to wrong (negative) returnvalues of overwritten modify function by
+                              // subdetector groups. A negative (-1) aktiveTime already indicate empty data maps!
+        double timeOfOldData = FindTimeForData(data);
+        // PrintDeadTimeMap();
+        if (timeOfOldData > -1) {   // if an older active data object is already present
+            if (fVerbose > 1) {
+                std::cout << " OldData found! " << timeOfOldData << std::endl;
+            }
+            if (fVerbose > 1) {
+                std::cout << "New Data: " << activeTime << " : " << data << std::endl;
+            }
+            double currentdeadtime = timeOfOldData;
+            FairTimeStamp* oldData;
+            bool dataFound = false;
 
-      // PrintDeadTimeMap();
-      for (DTMapIter it = fDeadTime_map.lower_bound(currentdeadtime); it != fDeadTime_map.upper_bound(currentdeadtime); it++) {
-        oldData = it->second;
-        if (fVerbose > 1) {
-          std::cout << "Check Data: " << it->first << " : " << oldData << std::endl;
-        }
-        if (oldData->equal(data)) {
-          if (fVerbose > 1) {
-            std::cout << " oldData == data " << std::endl;
-          }
-          if (fVerbose > 1) {
-            std::cout << it->first << " : " << it->second << std::endl;
-          }
-          fDeadTime_map.erase(it);
-          EraseDataFromDataMap(oldData);
-          dataFound = true;
-          break;
-        }
-      }
+            // PrintDeadTimeMap();
+            for (DTMapIter it = fDeadTime_map.lower_bound(currentdeadtime);
+                 it != fDeadTime_map.upper_bound(currentdeadtime);
+                 it++) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer prefix ++/-- operators for non-primitive types.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5323596)